attachments/forms.py

```diff
@@ -67,7 +67,11 @@ def formfield(self, prop, field_class=None, **kwargs):
             if defaults.get('required', False):
                 defaults['empty_label'] = None
         elif prop.data_type == 'boolean':
-            kwargs['initial'] = kwargs.get('initial', False) in (True, 'true')
+            kwargs['initial'] = kwargs.get('initial', False) in (True, 'true', ['on'], 'on')
+        elif prop.data_type == 'text' or prop.data_type == 'email':
+            initial_value = kwargs.get('initial')
+            if type(initial_value) is list and len(initial_value) == 1:
```

Comment:
This should be `isinstance(initial_value, (list, tuple))`. I'd probably also change the length check to `>= 1`. I think there might be some cleanup we can do to initial value handling in this method as a whole, but this is a fine start.

Reply:
Added a commit with the changes you suggest - using isinstance(), and updating the length check. I agree that there could be some additional cleanup in the method as a whole - should I submit a package enhancement issue, in the interest of getting this fix live?